### ISSUE TITLE
A server error occurs when logging in with a user that has no Role assigned

### ISSUE THREAD

**Issue by bbiagas** [directus/directus-8-legacy#2262]
Version: 8.6.2 

Abstract: When creating a new user via the Users Directory, the "Role" field is not required. If you create such a user, activate them, and attempt to log in with that user, the admin page will return a Server Error.

Related Issues: https://github.com/directus/app/issues/1476

Apologies if this is intended behavior. This same issue was discussed in passing in the above ticket, but this behavior seems unintuitive to me.

Conditions to Reproduce:

1) Bootstrap a minimal environment as detailed here: https://docs.directus.io/installation/docker.html

Steps to Reproduce:

1) Log in with default admin user
2) Visit http://localhost:8080/admin/#/_/users/+ to add a new user
3) Enter a value for "Email" (example: test@example.com)
4) Enter a value for "Password" (example: 12345)
5) Set "Status" to "Active"
6) Save the user
7) Log out of the admin user
8) Attempt to log in with the new user

Result:

A 401 Unauthorized is returned for the authentication request and the "Server Error" page is displayed

Expected Result:

Role should become a required field when a user is made "Active"

Additional Information:

This only occurs when creating a user from the User Directory. If you create a user directly on a Role that user correctly inherits that Role.

Limiting the Role check just to users who are marked "active" when creating/modifying users might prevent this case without harming the ability to have flexible user management. I'm sure there's a valid use case for adding users before you've decided what their role should be.

The logout link on the server error page also seems to be broken and doesn't kill the session and redirect back to the login page, which I can file a separate ticket for if one doesn't already exist. So the only way I've been able to cleanly recover from this error is to create a private browser window, log in with a known good user, and fix the role of the "broken" user. Once that is done the "broken" session will authenticate and everything is as expected.

Let me know if you need more information. Thanks for the awesome tools!
